File: src/test/java/com/eventr/config/TestConfig.java
package com.eventr.config;

import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;
import org.springframework.context.annotation.Profile;

import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.s3.S3Client;
import software.amazon.awssdk.services.dynamodb.DynamoDbClient;

import java.net.URI;

@TestConfiguration
@Profile("test")
public class TestConfig {

    @Bean
    @Primary
    public S3Client s3Client() {
        return S3Client.builder()
                .endpointOverride(URI.create("http://localhost:4566"))
                .credentialsProvider(StaticCredentialsProvider.create(
                        AwsBasicCredentials.create("test", "test")
                ))
                .region(Region.US_EAST_1)
                .forcePathStyle(true)
                .build();
    }

    @Bean
    @Primary
    public DynamoDbClient dynamoDbClient() {
        return DynamoDbClient.builder()
                .endpointOverride(URI.create("http://localhost:4566"))
                .credentialsProvider(StaticCredentialsProvider.create(
                        AwsBasicCredentials.create("test", "test")
                ))
                .region(Region.US_EAST_1)
                .build();
    }
}

File: src/test/kotlin/com/eventr/config/TestConfig.kt
package com.eventr.config

import com.eventr.service.QRCodeProcessingService
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary
import org.springframework.context.annotation.Profile
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import software.amazon.awssdk.services.dynamodb.DynamoDbClient
import software.amazon.awssdk.services.s3.S3Client
import org.mockito.Mockito

@TestConfiguration
@Profile("test")
class TestConfig {
    
    @Bean
    @Primary
    fun s3Client(): S3Client {
        return Mockito.mock(S3Client::class.java)
    }
    
    @Bean
    @Primary
    fun dynamoDbClient(): DynamoDbClient {
        return Mockito.mock(DynamoDbClient::class.java)
    }
    
    @Bean
    @Primary
    fun qrCodeProcessingService(): QRCodeProcessingService {
        return Mockito.mock(QRCodeProcessingService::class.java)
    }
    
    @Bean
    @Primary
    fun passwordEncoder(): BCryptPasswordEncoder {
        return BCryptPasswordEncoder()
    }
}